tests/improc/test_zogy.py

```diff
@@ -0,0 +1,436 @@
+        )
+        edge = 8  # do not trigger on stars too close to the edge
+        S = np.ones(zogy_score_corr.shape) * np.nan
+        S[edge:-edge, edge:-edge] = zogy_score_corr[edge:-edge, edge:-edge].copy()
+        B = truth2.background_instance
+        P = zogy_psf
+
+        for i, f in enumerate(fluxes):
+            x = y = int(pos[i])
+            c = S[y-edge:y+edge, x-edge:x+edge]
+            expected[j, i] = f * np.sqrt(np.sum(P ** 2) / B)
+            expected[j, i] *= np.sqrt(B / (B + f * np.sum(P ** 2)))  # add the Poisson noise from the source
+            measured[j, i] = np.nanmax(c)
+
+    mean = np.nanmean(measured, axis=0)
+    err = np.nanstd(measured, axis=0)
+    exp = expected[0, :]
+    p1 = np.polyfit(exp, mean, 1)
+
+    chi2 = np.sum((mean - np.polyval(p1, exp)) ** 2 / err ** 2)
+    assert chi2 / (len(fluxes) - 2) < 2.0  # the fit should be good
+    assert p1[0] > 0.8  # should be close to one, but we are losing about 15% S/N and I don't know why
+
+    # plt.errorbar(exp, mean, err, fmt='.', label='measured vs. expected')
+    # plt.plot(exp, np.polyval(p1, exp), 'r-', label=f'fit: {p1[0]:.3f}x + {p1[1]:.3f}')
+    # plt.xlabel('expected S/N')
+    # plt.ylabel('measured S/N')
+    # plt.legend()
+    # plt.show(block=True)
+
+
+@pytest.mark.flaky(max_runs=3)
+def test_subtraction_seeing_background():
+    num_stars = 300
+    sim = Simulator(
+        image_size_x=imsize,  # not too big, but allow some space for stars
+        background_std=0,  # keep background constant between images
+        background_minimum=0.01,  # allow very low background for the reference image
+        read_noise=0,  # make it easier to input the background noise (w/o source noise)
+        seeing_mean=2.0,  # make the seeing a little more pronounced
+        seeing_std=0,  # keep seeing constant between images
+        seeing_minimum=0.1,  # assume seeing never gets much better than this
+        transmission_std=0,  # keep transmission constant between images
+        pixel_qe_std=0,  # keep pixel QE constant between images
+        gain_std=0,  # keep pixel gain constant between images
+        saturation_limit=1e9,  # make it impossible to saturate
+        star_number=num_stars,
+    )
+    seeing_values = [2.0, 3.0, 5.0]
+    background_values = [1.0, 3.0, 10.0]
+
+    fluxes = [300, 500, 1000]
+    pos = np.linspace(0, imsize, len(fluxes) + 1, endpoint=False)[1:]
+
+    sim.make_stars()
+
+    # verify no stars are on top of our injection positions
+    # ZOGY still detects the new source on top of the star, but the S/N is reduced
+    for p in pos:
+        for i in range(len(sim.stars.star_mean_fluxes)):
+            if abs(p - sim.stars.star_mean_x_pos[i]) < 10:
+                sim.stars.star_mean_x_pos[i] += 15  # bump this star out of the way
+            if abs(p - sim.stars.star_mean_y_pos[i]) < 10:
+                sim.stars.star_mean_y_pos[i] += 15  # bump this star out of the way
+
+    # beware the 4-loop!
```

Comment:
Fortunately, 3^4 is only 81

Reply:
It is more of a warning to myself not to get to generous with the list of parameters to test.